src/ecs/src/dynvec.rs

```diff
@@ -0,0 +1,73 @@
+//! A runtime-typed vector. All DynVecs are of the same type, regardless of their elements' types.
+
+use std::any::{Any, TypeId};
+use std::mem::{transmute, size_of};
+use std::ops::{Index, IndexMut};
+
+#[derive(Debug)]
+pub struct DynVec {
+    vec: Vec<u8>,
+    unused: Vec<usize>,
+    t: TypeId,
+    size: usize,
+}
+
+impl DynVec {
+    /// Creates a new dynamically typed vector of type T
+    pub fn new<T: Any>() -> DynVec {
+        DynVec {
+            vec: vec![],
```

Comment:
I think the usage of `vec![]` should be replaced with `Vec::new()`, to follow idiomatic practice.